### PR TITLE
IStateTransfe change to unqiue_ptr

### PR DESCRIPTION
**Pull request by titusen**
Replica.cpp is leaking memory due to undeleted memory.
Change from "*" pointer to "unique_ptr"

apollo tests passed

### CODE REVIEW

File: kvbc/include/Replica.h [hunk 1]
```diff
@@ -243,7 +243,7 @@ class Replica : public IReplica,
   const bftEngine::ReplicaConfig &replicaConfig_;
   bftEngine::IReplica::IReplicaPtr m_replicaPtr = nullptr;
   std::shared_ptr<ICommandsHandler> m_cmdHandler = nullptr;
-  bftEngine::IStateTransfer *m_stateTransfer = nullptr;
+  std::unique_ptr<bftEngine::IStateTransfer> m_stateTransfer;
```
Comment (glevkovich) [inline kvbc/include/Replica.h:246]: - The header of the PR looks wrong - you are replacing State transfer pointer with a raw pointer.
- Why are you merging into master? we still have a dev branch.

File: kvbc/src/Replica.cpp [hunk 2]
```diff
@@ -600,7 +600,7 @@ Replica::Replica(ICommunication *comm,
   m_dbSet.dataDBClient->setAggregator(aggregator_);
   m_dbSet.metadataDBClient->setAggregator(aggregator_);
   auto stKeyManipulator = std::shared_ptr<storage::ISTKeyManipulator>{storageFactory->newSTKeyManipulator()};
-  m_stateTransfer = bftEngine::bcst::create(stConfig, this, m_metadataDBClient, stKeyManipulator, aggregator_);
+  m_stateTransfer.reset(bftEngine::bcst::create(stConfig, this, m_metadataDBClient, stKeyManipulator, aggregator_));
```
Comment (glevkovich) [inline kvbc/src/Replica.cpp:603]: So it looks like ST was nver called since we never called it's ctor, right?

**Review by glevkovich — CHANGES_REQUESTED** [review]

File: kvbc/src/Replica.cpp (hunk 2 above)
Comment (titusen) [reply kvbc/src/Replica.cpp:603]: it is called later with "*m_stateTransfer", so the syntax persists.

File: kvbc/include/Replica.h (hunk 1 above)
Comment (glevkovich) [reply kvbc/include/Replica.h:246]: > No, other way round "change **to** unique_ptr"

@titusen 
Suggested header for the PR:
"Convert m_stateTransfer to a unqiue_ptr"

**Pull request closed without merging**